docs/src/concepts/index_ivfpq.md

```diff
@@ -0,0 +1,78 @@
+
+## IVF-PQ
+
+IVF-PQ is a composite index that combines inverted file index (IVF) and product quantization (PQ). The implementation in LanceDB provides several parameters to fine-tune the index's size, query throughput, latency and recall, which are described later in this section.
+
+### Product quantization
+
+Quantization is a compression technique used to reduce the dimensionality of an embedding to speed up search. 
+
+![](../assets/ivfpq_quantization.webp)
+
+Product quantization (PQ) works by dividing a large, high-dimensional vector of size into equally sized subvectors. Each subvector is assigned a "reproduction value" that maps to the nearest centroid of points for that subvector. The reproduction values are then assigned to a codebook using unique IDs, which can be used to reconstruct the original vector.
+
+![](../assets/ivfpq_pq_desc.webp)
+
+It's important to remember that quantization is a *lossy process*, i.e., the reconstructed vector is not identical to the original vector. This results in a trade-off between the size of the index and the accuracy of the search results.
+
+As an example, consider a starting 128-dimensional vector consisting of 32-bit floats. Quantizing it to an 8-bit integer vector with **only 8 dimensions**, we can cause a significant reduction in memory requirements.
+
+!!! example "Effect of quantization"
+
+    Original: `128 × 32 = 4096` bits  
+    Quantized: `8 × 8 = 64` bits  
+
+    Quantization results in a **64x** reduction in memory requirements for each vector in the index, which is substantial.
+
+### Inverted file index
+
+While PQ helps with reducing the size of the index, IVF primarily addresses search performance. The primary purpose of an inverted file index is to facilitate rapid and effective nearest neighbor search by narrowing down the search space.
+
+In IVF, the PQ vector space is divided into *Voronoi cells*, which are essentially partitions that consist of all the points in the space that are within a threshold distance of the given region's seed point. These seed points are used to create an inverted index that correlates each centroid with a list of vectors in the space, allowing a search to be restricted to just a subset of vectors in the index.
+
+![](../assets/ivfpq_ivf_desc.webp)
+
+During query time, depending on where the query lands in vector space, it may be close to the border of multiple Voronoi cells, which could make the top-k results ambiguous and span across multiple cells. To address this, the IVF-PQ introduces the `nprobe` parameter, which controls the number of Voronoi cells to search during a query. The higher the `nprobe`, the more accurate the results, but the slower the query.
+
+![](../assets/ivfpq_query_vector.webp)
+
+## Putting it all together
+
+We can combine the above concepts to understand how to build and query an IVF-PQ index in LanceDB.
+
+### Construct index
```

Comment:
I think we could say a little more in this part, telling how we decided the `num_partitions` and `num_sub_vectors` parameters. `num_partitions` is usually chosen to target a particular number of vectors per partition. `num_sub_vectors` will likely depend on the desired recall and dimensionality of the vector.

Reply:
I've added the details on how we choose these values and some best practices in the FAQs section from the old docs.